### PR TITLE
In Boutiques, fall back on SingleFile when output file type suggests nil type.

### PR DESCRIPTION
**Pull request by glatard**

### CODE REVIEW

File: BrainPortal/lib/cbrain_task_generators/templates/bourreau.rb.erb [hunk 1]
```diff
@@ -299,10 +299,10 @@ class CbrainTask::<%= name %> < ClusterTask
...
         self.addlog("Saving result file #{path}")
         name = File.basename(path)
-
-        output = safe_userfile_find_or_new((
-          File.directory?(path) ? FileCollection : Userfile.suggested_file_type(name)
-        ), :name => name)
+        
+	userfile_class = Userfile.suggested_file_type(name)
+        userfile_class ||= ( File.directory?(path) ? FileCollection : SingleFile )
+        output = safe_userfile_find_or_new(userfile_class, :name => name)
```
Comment (remibernard) [inline BrainPortal/lib/cbrain_task_generators/templates/bourreau.rb.erb]: Nitpick; it seems like the indentation/whitespace is kinda wonky; line 302 has EOL whitespace, 303 has incorrect indentation and 304 has extra whitespace around the parentheses for the ternary operator (unlike anywhere else in the template). Shouldn't it look like:
```ruby
userfile_class   = Userfile.suggested_file_type(name)
userfile_class ||= (File.directory?(path) ? FileCollection : SingleFile)
output = safe_userfile_find_or_new(userfile_class, :name => name)
```

**Comment by natacha-beck** [comment]
I will merge it. Does it close any issues ?

**Comment by glatard** [comment]
No, it doesn't close any issue. It solves a problem that I just noticed on a tool I am working on.

File: BrainPortal/lib/cbrain_task_generators/templates/bourreau.rb.erb (hunk 1 above)
Comment (driusan) [inline BrainPortal/lib/cbrain_task_generators/templates/bourreau.rb.erb]: On the LORIS team we've gotten rid of such issues (and the need to point them out) by having Travis run a static analysis tool that checks for style issues like inconsistent whitespace and fails the build if it finds them.

I'm sure Ruby must have some kind of linter that you can use.

**Pull request merged**